src/debug_menu.cpp

```diff
@@ -121,99 +128,133 @@ class mission_debug
-        { DEBUG_SET_AUTOMOVE, true, 'a', _( "Set automove route" ) }
+    return ( display_all_entries || ( !display_all_entries && e.second ) );
+};
+
+static int player_uilist( bool display_all_entries = true )
+{
+    /* Note: the second boolean indicates whether the menu should always be shown (true) or not (false).
+     * For example, if we show the debug menu on the main menu and we want Info to be always displayed, we set it to true.
+     * This also depends if the top menu is shown or not, in which case,the booleans have no effect if the top menu is not shown.
+     */
+    const std::vector<uilist_entry_pair_t> uilist_initializer = {
+        {uilist_entry( DEBUG_MUTATE, true, 'M', _( "Mutate" ) ), true },
+        { uilist_entry( DEBUG_CHANGE_SKILLS, true, 's', _( "Change all skills" ) ), true },
+        {uilist_entry( DEBUG_LEARN_MA, true, 'l', _( "Learn all melee styles" ) ), true },
+        {uilist_entry( DEBUG_UNLOCK_RECIPES, true, 'r', _( "Unlock all recipes" ) ), true },
+        {uilist_entry( DEBUG_EDIT_PLAYER, true, 'p', _( "Edit player/NPC" ) ), true },
+        {uilist_entry( DEBUG_DAMAGE_SELF, true, 'd', _( "Damage self" ) ), true },
+        {uilist_entry( DEBUG_SET_AUTOMOVE, true, 'a', _( "Set automove route" ) ), true },
     };
 
-    return uilist( _( "Player..." ), uilist_initializer );
+    const auto filter = [&display_all_entries]( const uilist_entry_pair_t &e ) -> bool { return display_selector( display_all_entries, e ); };
+    return uilist( _( "Player..." ), uilist_initializer, filter );
 }
 
-static int info_uilist()
+static int info_uilist( bool display_all_entries = true )
 {
-    const std::vector<uilist_entry> uilist_initializer = {
-        uilist_entry( DEBUG_GAME_STATE, true, 'g', _( "Check game state" ) ),
-        uilist_entry( DEBUG_DISPLAY_HORDES, true, 'h', _( "Display hordes" ) ),
-        uilist_entry( DEBUG_TEST_IT_GROUP, true, 'i', _( "Test item group" ) ),
-        uilist_entry( DEBUG_SHOW_SOUND, true, 's', _( "Show sound clustering" ) ),
-        uilist_entry( DEBUG_DISPLAY_WEATHER, true, 'w', _( "Display weather" ) ),
-        uilist_entry( DEBUG_DISPLAY_SCENTS, true, 'S', _( "Display overmap scents" ) ),
-        uilist_entry( DEBUG_SHOW_MUT_CAT, true, 'm', _( "Show mutation category levels" ) ),
-        uilist_entry( DEBUG_BENCHMARK, true, 'b', _( "Draw benchmark (X seconds)" ) ),
-        uilist_entry( DEBUG_TRAIT_GROUP, true, 't', _( "Test trait group" ) ),
-        uilist_entry( DEBUG_SHOW_MSG, true, 'd', _( "Show debug message" ) ),
-        uilist_entry( DEBUG_CRASH_GAME, true, 'C', _( "Crash game (test crash handling)" ) ),
-        uilist_entry( DEBUG_DISPLAY_NPC_PATH, true, 'n', _( "Toggle NPC pathfinding on map" ) ),
-        uilist_entry( DEBUG_TEST_WEATHER, true, 'W', _( "Test weather" ) )
+    const std::vector<uilist_entry_pair_t> uilist_initializer = {
+        {uilist_entry( DEBUG_GAME_STATE, true, 'g', _( "Check game state" ) ), false},
```

Comment:
```suggestion
        { uilist_entry( DEBUG_GAME_STATE, true, 'g', _( "Check game state" ) ), false },
```
(and similarly below)